index.js

```diff
@@ -995,21 +1007,29 @@ module.exports = function (log, indexesPath) {
         )
       }),
 
-      // Create missing indexes, if any
+      // some indexes might have failed to load
+      push.asyncMap((_, next) => {
+        opsMissingIdx = detectMissingAndLazyIndexes(operation)[0]
```

Comment:
```suggestion
        [opsMissingIdx] = detectMissingAndLazyIndexes(operation)
```

Reply:
It's a bit unfortunate that the first array `opsMissingIdx` is calculated but never used, because it will always be re-calculated here. Maybe we could split up `detectMissingAndLazyIndexes` to `detectMissingIndexes` and `detectLazyIndexes`? (I can do that if you want)

Reply:
Yeah I was thinking of splitting that function up because we only use missing on second go. Feel free to fix that :)